samza-api/src/main/java/org/apache/samza/application/StreamApplication.java

```diff
@@ -53,45 +47,29 @@
  *     CommandLine cmdLine = new CommandLine();
  *     Config config = cmdLine.loadConfig(cmdLine.parser().parse(args));
  *     PageViewCounter app = new PageViewCounter();
- *     LocalApplicationRunner runner = new LocalApplicationRunner(config);
- *     runner.run(app);
+ *     ApplicationRunner runner = ApplicationRunners.getApplicationRunner(new PageViewCounter(), config);
+ *     runner.run();
  *     runner.waitForFinish();
  *   }
  * }</pre>
  *
  * <p>
- * Implementation Notes: Currently StreamApplications are wrapped in a {@link StreamTask} during execution.
- * A new StreamApplication instance will be created and initialized with a user-defined {@link StreamGraph}
- * when planning the execution. The {@link StreamGraph} and the functions implemented for transforms are required to
- * be serializable. The execution planner will generate a serialized DAG which will be deserialized in each {@link StreamTask}
- * instance used for processing incoming messages. Execution is synchronous and thread-safe within each {@link StreamTask}.
+ * Implementation Notes: Currently StreamApplications are wrapped in a {@link org.apache.samza.task.StreamTask} during
+ * execution. A new {@link StreamAppDescriptor} instance will be created and described by the user-defined
```

Comment:
Removed all clarification on the serializability of descriptors, since descriptors should be provided by system implementers. Agree that users should only care about user-defined transformation functions to be serializable.